app/Module.scala

```diff
@@ -84,18 +91,28 @@ class Module(environment: Environment, configuration: Configuration) extends Abs
     @Inject() restRepository: RestRepository,
     unitLinksRepositoryConfig: HBaseRestUnitLinksRepositoryConfig,
     rowMapper: RowMapper[UnitLinksNoPeriod],
-    @Named("unit-link") unitLinkRegisterService: UnitRegisterService
+    @Named(UnitLink) unitLinkRegisterService: UnitRegisterService
   ): UnitLinksRepository =
     new HBaseRestUnitLinksRepository(restRepository, unitLinksRepositoryConfig, rowMapper, unitLinkRegisterService)
 
   @Provides
   def providesPatchService(
     @Inject() unitLinksRepository: UnitLinksRepository,
-    @Named("vat") vatRegisterService: UnitRegisterService,
-    @Named("unit-link") unitLinkRegisterService: UnitRegisterService
+    @Named(Vat) vatRegisterService: UnitRegisterService,
+    @Named(Paye) payeRegisterService: UnitRegisterService,
+    @Named(UnitLink) unitLinkRegisterService: UnitRegisterService
   ): PatchService = {
-    val vatUnitLinkPatchService = new VatUnitLinkPatchService(unitLinksRepository, unitLinkRegisterService)
-    val legalUnitLinkPatchService = new LegalUnitLinkPatchService(unitLinksRepository, vatRegisterService)
-    new UnitLinksPatchService(vatUnitLinkPatchService, legalUnitLinkPatchService)
+    val adminDataUnitLinkPatchService = new AdminUnitLinkPatchService(unitLinksRepository, unitLinkRegisterService)
+    val adminDataRegisterService = new CompositeAdminUnitRegisterService(vatRegisterService, payeRegisterService)
+    val legalUnitLinkPatchService = new LegalUnitLinkPatchService(unitLinksRepository, adminDataRegisterService)
+    new UnitLinksPatchService(adminDataUnitLinkPatchService, legalUnitLinkPatchService)
   }
 }
+
+private object Module {
+  object Names {
+    final val Paye = "paye"
```

Comment:
Just curious, why the `final`? As I understand it, `final` is to prevent overriding, but would `override` ever be used on an `object`?

Reply:
I have added comments explaining this in previous usage - but left it out this time as I thought everyone had seen them by now.  I guess I was wrong ... ;-)

We are using the values in ```Names``` as constants in Guice injection ```@Named``` annotations.  The annotation processor requires these values to be "constant", and unfortunately Scala's ```val``` is not sufficient.  They have to be explicitly declared ```final``` for the annotation processor to be happy.